conifer/backends/cpp/include/conifer.h

```diff
@@ -91,14 +100,21 @@ class DecisionTree{
                    [](double t) -> T { return (T) t; });
     std::transform(value.begin(), value.end(), std::back_inserter(value_),
                    [](double v) -> U { return (U) v; });
+    // Load weight vector of arrays, multidimensional so no nice one liner?
+    for (int i = 0; i < weight.size(); i++) {
+      weight_.push_back(std::vector<W>());
+        for (int j = 0; j < weight[i].size(); j++) {
+            weight_[i].push_back((W) weight[i][j]);
+        }
+    }
   }
 
   // Define how to read this class to/from JSON
-  NLOHMANN_DEFINE_TYPE_INTRUSIVE(DecisionTree, feature, children_left, children_right, threshold, value);
+  NLOHMANN_DEFINE_TYPE_INTRUSIVE(DecisionTree, feature, weight, children_left, children_right, threshold, value);
 
 }; // class DecisionTree
 
-template<class T, class U, bool useAddTree = false>
+template<class T, class U, class W, bool useAddTree = false>
```

Comment:
Maybe now the number of template arguments expands, we could consolidate with a configuration `struct`. Something like this, that the user would need to provide (and that we could auto-generate with the C++ backend).

```
struct BDTConfig {
    // Type aliases
    typedef float threshold_t;
    typedef float precision_t;
    typedef float weight_t;
    typedef float score_t;

    // Fields
    bool useAddTree;
};
```